terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/internal/TerminalService.java

```diff
@@ -311,11 +302,11 @@ private IStatus doRun(TerminalViewId tvid, String title, ITerminalConnector conn
 	@Override
 	public CompletableFuture<?> closeConsole(Map<String, Object> properties) {
 		Assert.isNotNull(properties);
-		return executeServiceOperation(properties, new TerminalServiceRunnable() {
-			@Override
-			public IStatus run(TerminalViewId tvid, String title, ITerminalConnector connector, Object data) {
-				consoleViewManager.closeConsole(tvid, title, connector, data);
-				return Status.OK_STATUS;
+		var future = executeServiceOperation(properties,
```

Comment:
I find var a bit hard to read often because you need to now derive the type of the variable and as we do not have a complex or varing eeturn type I would better just use 
```suggestion
		CompletableFuture<?> future = executeServiceOperation(properties,
```

Reply:
I just realized I made a mistake and should not have added the whenComplete in this location at all because we should leave the caller to handle the exceptions since we are returning future here.

BTW TerminalService's `closeConsole` and `terminateConsole` are not actually called from any code I know of